controllers/vspherecluster_reconciler.go

```diff
@@ -440,7 +440,7 @@ func (r clusterReconciler) reconcileVSphereClusterWhenAPIServerIsOnline(clusterC
 	go func() {
 		// Block until the target API server is online.
 		clusterCtx.Logger.Info("start polling API server for online check")
-		wait.PollUntilContextCancel(context.Background(), time.Second*1, true, func(context.Context) (bool, error) { return r.isAPIServerOnline(clusterCtx), nil }) //nolint:errcheck
+		wait.PollUntilContextCancel(ctx, time.Second*1, true, func(context.Context) (bool, error) { return r.isAPIServerOnline(ctx, clusterCtx), nil }) //nolint:errcheck
```

Comment:
Let's keep background here and in l.455

The problem is that this go routine continues to run after the reconcile is done. So we don't want to cancel them

Reply:
Thanks for catching it!